src/util/unicode.cpp

```diff
@@ -20,6 +19,24 @@ Author: Daniel Kroening, kroening@kroening.com
 
 /*******************************************************************\
 
+Function: is_little_endian_arch
+
+  Inputs:
+
+ Outputs: True if the architecture is little_endian
+
+ Purpose: Determine endianness of the architecture
+
+\*******************************************************************/
+
+bool is_little_endian_arch()
+{
+  uint16_t i=1;
```

Comment:
There are architectures where bytes (or characters?) aren't 8 bits wide (but 16 instead). That said: does this have to be a compile-time or rather a runtime check? Or is it even a target-analysis architecture property?